cmd/disk-cache-fs.go

```diff
@@ -72,9 +77,23 @@ func newCacheFSObjects(dir string, expiry int, maxDiskUsagePct int) (*cacheFSObj
 	if expiry == 0 {
 		expiry = globalCacheExpiry
 	}
-	var cacheFS cacheFSObjects
-	fsObjects := obj.(*FSObjects)
-	cacheFS = cacheFSObjects{
+
+	// Initialize fs objects.
+	fsObjects := &FSObjects{
+		fsPath:       dir,
+		metaJSONFile: cacheMetaJSONFile,
```

Comment:
if you want to use it like this then you can change the signature of `newFSObjects` and avoid taking in `fsMetaJSON` as an argument.